packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -653,40 +610,48 @@ export const makeInnerVault = (
       AmountMath.isEmpty(state.debtSnapshot),
       X`vault must be empty initially`,
     );
-    const oldDebt = getCurrentDebt();
-    const oldCollateral = getCollateralAmount();
-    trace('initVaultKit start: collateral', { oldDebt, oldCollateral });
+    // TODO should this be simplified to know that the oldDebt mut be empty?
+    const debtPre = getCurrentDebt();
+    const collateralPre = getCollateralAmount();
+    trace('initVaultKit start: collateral', { debtPre, collateralPre });
 
     // get the payout to provide access to the collateral if the
     // contract abandons
     const {
-      give: { Collateral: collateralAmount },
-      want: { RUN: wantedRun },
+      give: { Collateral: giveCollateral },
+      want: { RUN: wantRUN },
     } = seat.getProposal();
 
     // todo trigger process() check right away, in case the price dropped while we ran
+    const {
+      newDebt: newDebtPre,
+      fee,
+      toMint,
+    } = loanFee(debtPre, emptyDebt, wantRUN);
+    assert(
+      !AmountMath.isEmpty(fee),
+      X`loan requested (${wantRUN}) is too small; cannot accrue interest`,
+    );
+    assert(AmountMath.isEqual(newDebtPre, toMint), X`loan fee mismatch`);
```

Comment:
took me a second to see why these values must be equal.
```suggestion
    assert(AmountMath.isEqual(newDebtPre, toMint), X`fee mismatch for new vault`);
```